### DISCUSSION TITLE
flash messages

### DISCUSSION THREAD

**Discussion by siddhantk232** [fastn-stack/fastn#1725] [Ideas & RFCs]
A port of this would be interesting: https://docs.djangoproject.com/en/5.0/ref/contrib/messages/

**Comment by amitu** [comment]
Off late ["toast ui"](https://getbootstrap.com/docs/4.2/components/toasts/) is [becoming](https://ellie-app.com/mj4LGXqstwa1) [a](https://svelte-french-toast.com) [trend](https://react-hot-toast.com). Toast stays on the current page, and shows some sort of confirmation. This is useful when you can not show the effect on the current page (eg posting a tweet, you can show the timeline with tweet on top if you are on the right page). But if you are on a different page, where the end result can not show up, you have two options: a. same-page-toast, or b. redirect-and-message. 

The message framework is kind of for redirect and message. 

The django implementation, storing the message in the database, has one issue, that it can show up in the wrong place. Tho one can argue it is the right thing. Like if you confirmed your email on mobile, but say the next page did not load, so message is still in db, and you then open desktop, and you will see the message. This can be good, but only if the message you show has enough context, so user does not get confused.

The other possibility is to store the "flash" message in the session store. We anyways do all our actions via the "action" way, where there is an ajax request, which returns redirect on success. This response can include the flash message, and our http.js can handle it in a generic way, by capture it and storing it in session store. With session store, the message will only show in the same tab where the action was carried out.

**Reply by amitu** [reply]
Maybe we can return our action response "protocol" / "contract" to include extra json keys, like instead of redirect you may just send flash, and the message will be "toasted". If you send both flash and redirect, it will be stored in session store and toasted on the next page.

 And since the page would like to control how to display the toast, we should create some `-- optional string ftd.flash:` or something, so the page can decide how to show this message. Of course it being a string is quite limiting, we will be forced to use innerHTML like stuff, and not use full power of ftd to show the toast message (say you had a user view-tweet-flash ftd component, which needs proper data, not just a string, how do we engineer this?.
